addons/hr_recruitment/models/hr_job.py

```diff
@@ -108,6 +108,17 @@ def create(self, vals):
         vals['favorite_user_ids'] = vals.get('favorite_user_ids', []) + [(4, self.env.uid)]
         return super(Job, self).create(vals)
 
+    @api.model
+    def search_panel_select_range(self, field_name):
+        if field_name == 'user_id':
+            users = self.env['hr.job'].search([('user_id', '!=', None)]).mapped('user_id')
+            available_users = self.env['res.users'].search([('id', 'in', users.ids)], limit=200).mapped(lambda user: { 'id': user.id, 'display_name': user.name})
```

Comment:
instead of `200` you should use:

```py
from odoo.addons.web.models.models import SEARCH_PANEL_LIMIT
```
instead of:

```py
.mapped(lambda user: { 'id': user.id, 'display_name': user.name})
```

I think you can do:

```py
.read(['display_name'])
```